src/coreclr/jit/codegenxarch.cpp

```diff
@@ -738,12 +739,18 @@ void CodeGen::genCodeForNegNot(GenTree* tree)
     {
         GenTree* operand = tree->gtGetOp1();
         assert(operand->isUsedFromReg());
-        regNumber operandReg = genConsumeReg(operand);
+        regNumber   operandReg = genConsumeReg(operand);
+        instruction ins        = genGetInsForOper(tree->OperGet(), targetType);
 
-        inst_Mov(targetType, targetReg, operandReg, /* canSkip */ true);
-
-        instruction ins = genGetInsForOper(tree->OperGet(), targetType);
-        inst_RV(ins, targetReg, targetType);
+        if (GetEmitter()->DoJitUseApxNDD(ins) && (targetReg != operandReg))
+        {
+            GetEmitter()->emitIns_R_R(ins, emitTypeSize(operand), targetReg, operandReg, INS_OPTS_EVEX_nd);
+        }
+        else
+        {
+            inst_Mov(targetType, targetReg, operandReg, /* canSkip */ true);
+            inst_RV(ins, targetReg, targetType);
+        }
```

Comment:
This general pattern is repeated quite a lot (with some variations), so I wonder if we should have a helper like I added for SIMD.

For example, we have `emitIns_SIMD_R_R_R` which looks like: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/emitxarch.cpp#L8855-L8880

This lets us correctly represent any SIMD `dst = src1 op src2` operation given the raw registers and then internally handles the RMW consideration, so that the rest of codegen can remain simpler and more readable.

Reply:
In this case, for example, it seems like we "could" have simplified this down to something like:
```
GetEmitter()->emitIns_BASE_R_R(ins, emitTypeSize(operand), targetReg, operandReg);
```

and than had this helper make the distinction of handling `APX`, `NDD`, inserting the `Mov` for the regular case; etc

Presumably this would also make the diffs for other APX support much simpler as well, since we have fewer centralized helpers to update.